File: sportsbook-app/betting-service/src/main/java/com/sportsbook/bettingservice/client/OddsRestClient.java
package com.sportsbook.bettingservice.client;

import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;
import java.util.Map;

@Component
public class OddsRestClient {
    private final RestTemplate restTemplate = new RestTemplate();

    public Map<String, Double> getOdds(String matchId) {
        return restTemplate.getForObject("http://localhost:8083/api/odds/" + matchId, Map.class);
    }
}

File: sportsbook-app/betting-service/src/main/kotlin/com/sportsbook/bettingservice/client/OddsRestClient.kt
package com.sportsbook.bettingservice.client

import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class OddsRestClient {
    private val restTemplate = RestTemplate()

    fun getOdds(matchId: String): Map<String, Double> {
        val raw = restTemplate.getForObject(
            "http://localhost:8083/api/odds/$matchId",
            Map::class.java,
        ) ?: emptyMap<Any, Any>()
        return raw.entries.associate { (key, value) ->
            key.toString() to (value as Number).toDouble()
        }
    }
}
